src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -1378,11 +1378,17 @@ public SQLServerDataSource getDatasource() {
   }
 
   private SQLServerDataSource getDatasource(String databaseName) {
```

Comment:
We call `getDatasource` all over, and these retry settings are pretty high -- looks like we'll max out at 5 minutes?

Would it be more appropriate to only obtain the datasource with these retries on initialization only, since that's where we're running into the timeout issues?

Reply:
I was wondering about that.  I _think_ it's ok.  Especially since the thing in our case that really makes it take a while is warming and that's a rare thing

Reply:
I _think_ I've seen these issues elsewhere, so I think it's fine for there to be a retry on any query. But, maybe we should shorten the timeout?

Reply:
Sounds good!

In that case, I'd advocate for moving these overrides to be property-driven or minimally defined at the top level of the file with docstrings.  In the past I've found those call-outs helpful when overriding default values.

Reply:
How short do you think the timeout should be @snf2ye? Maybe 2 minutes instead of 5?

Reply:
Sadly, they don't really give any indication on time to wait:
https://learn.microsoft.com/sr-cyrl-rs/sql/relational-databases/errors-events/mssqlserver-42109-database-engine-error?view=sql-server-ver15

or
https://learn.microsoft.com/sr-cyrl-rs/sql/relational-databases/errors-events/mssqlserver-42109-database-engine-error?view=sql-server-ver15

5 minutes might actually be ok (it doesn't mean it'll use all of it)

I also agree with @okotsopoulos's point about making it a config value (probably on the bio.terra.service.resourcemanagement.azure.AzureResourceConfiguration.Synapse object)

Reply:
Keeping it 5 minutes but adding config values makes a lot of sense!

Reply:
Awesome, looks great!